core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/CarbonUtils.java

```diff
@@ -1266,9 +1269,12 @@ public static String getGhostMetafileDir(AxisConfiguration axisConfig) {
     public static Deployer getDeployer(String className) throws CarbonException {
         Deployer deployer;
         try {
-            Class deployerClass = Class.forName(className);
-            deployer = (Deployer) deployerClass.newInstance();
-
+            if (allowedDeployers.contains(className)) {
```

Comment:
org.apache.axis2.deployment.ServiceDeployer
is this the only deployer we allow?

Reply:
The usage of the method was only in this repo. The only allowed class was this.